sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -92,6 +94,38 @@ def _validate_parameters(self):
                 'multiclass classification.'
                 )
 
+    def _check_categories(self, n_features, X_orig):
+        """Check and validate categories params in X"""
+        if self.categorical is None:
+            self.categorical_features_ = None
+            return
+
+        error_msg = ("categorical must be an array-like of bool with shape "
+                     "(n_features,) or 'pandas'")
```

Comment:
Maybe we can be nice and also accept an array-like of names and an array-like of indices? The mapping should be straighforward.

Reply:
Since it is also possible for ColumnTransformer, I agree that it would nice to also accept it here.